e2e-env/cloudsync-mock/src/lib/ws-server.ts

```diff
@@ -1,19 +1,32 @@
 import {WebSocketServer} from 'ws';
 import {Request} from "express";
 
+const MAX_AGE = 30 * 1000;
+const MAX_SIZE = 1000;
+
 /**
  * Websocket server used by [mock-probe.ts] to wait for and inspect requests during testing.
  */
 export class WsServer {
   private server: WebSocketServer;
 
+  // keep some history in a FIFO in case probe connects after ps_eventbus request
+  private history: { timestamp: number, data: unknown }[] = []
```

Comment:
this is necessary on slow environments, such as github ci or when my laptop is unplugged...